test/integration/singlecluster/scheduler/scheduler_test.go

```diff
@@ -2450,4 +2450,139 @@ var _ = ginkgo.Describe("Scheduler", func() {
 			}, util.Timeout, util.Interval).Should(gomega.Succeed())
 		})
 	})
+
+	ginkgo.When("Handling multiple workloads with incompatible flavors in a ClusterQueue", func() {
```

Comment:
I believe it could be tested in unit test reliably, in scheduler_test.go, `TestSchedule`. 

The scheduler tests are much faster to run, easier to debug, and they can also cover this part functionally (I believe). 

wdyt?

Reply:
To be clear - I'm not 100% sure, maybe there are some aspects of the scenario which require multiple scheduling cycles. In that case, it is possible it makes sense to test on both levels - unit and integration.

Reply:
Actually, the integration test also passes even if the `attemptedFlavorIdx` is moved back to: https://github.com/kubernetes-sigs/kueue/pull/2407/files#diff-0870ec2c0ea330a7d087a6ef599905bf9b7164a0dd889d1b00ba17c31a1bfd05L430, so I don't think the test in the current form meets the requirements.